### PR TITLE
RT/JDJ/Rule 5-1

### PR DESCRIPTION
**Pull request by JacksonJ-KC**

### CODE REVIEW

File: rct229/ruletest_engine/ruletest_jsons/scripts/excel_to_test_json.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 update_unit_convention_record(spreadsheet_name, sheet_name)
 
 # Check generated master JSON against latest schema
-json_dir = os.path.join(os.path.dirname(__file__), "..", rule_set)
+json_dir = os.path.join(os.path.dirname(__file__), "../../../../../../..", rule_set)
```
Comment (yunjoonjung-PNNL) [inline rct229/ruletest_engine/ruletest_jsons/scripts/excel_to_test_json.py:33]: Could you check if this change was intended?

Comment (JacksonJ-KC) [reply rct229/ruletest_engine/ruletest_jsons/scripts/excel_to_test_json.py]: thanks, did not intend to commit this change

**Comment by weilixu** [comment]
Is this PR for the test json or RDS? - it seems include the RDS but the RDS was reviewed in the other PR.

**Comment by JacksonJ-KC** [comment]
> Is this PR for the test json or RDS? - it seems include the RDS but the RDS was reviewed in the other PR.

Sorry for the confusion. I wanted to be able to reference the unmerged RDS from within my IDE so I started this branch for TCDs from the branch for the RDS. Does this cause problems? I can close this PR and create a fresh branch from develop with the test json only.

**Comment by weilixu** [comment]
> > Is this PR for the test json or RDS? - it seems include the RDS but the RDS was reviewed in the other PR.
> 
> Sorry for the confusion. I wanted to be able to reference the unmerged RDS from within my IDE so I started this branch for TCDs from the branch for the RDS. Does this cause problems? I can close this PR and create a fresh branch from develop with the test json only.

It looked fine on my branch. I was confused and thought you were developing another Rule 5-1. Just want to confirm. You should be fine with what you have in this PR, Thank you for the clarification.

**Comment by JacksonJ-KC** [comment]
@weilixu @yunjoonjung-PNNL I gave this rule test json a quick review, and corrected an error where I did not include output_instance elements. Let me know if you run into any issues

**Comment by yunjoonjung-PNNL** [comment]
Hi Jackson, Great work! I have a question and a few comments.

**Question**
1. RDS in the develop and your branch is a bit different. To me,  the RDS in your branch is more up-to-date. Could we merge your RDS to the develop branch in a new branch? This is because I also made a few updates in the RDS. If I pull this branch, it'd be a bit confusing.
image

**Comments**
1. It seems like the TCD in this branch isn't valid. Could you check this out?
2. I believe User/Proposed should have the ```type``` key similar to baseline. 
3. We need the ```tilt``` key under the ```surfaces``` b/c the key is required in the ```get_opaque_surface_type``` function.

Please let me know if any clarification is needed!

**Comment by JacksonJ-KC** [comment]
@yunjoonjung-PNNL I made the updates for RMD type and surface tilt, and verified that the json passes validation.

**Comment by yunjoonjung-PNNL** [comment]
> @yunjoonjung-PNNL I made the updates for RMD type and surface tilt, and verified that the json passes validation.

Much appreciated Jackson. I'll test it out sometime tomorrow and leave any comment if I have any questions.

**Comment by JacksonJ-KC** [comment]
> Hi @JacksonJ-KC , Great work! I have a question and a few comments.
> 
> **Question**
> 
> 1. RDS in the develop and your branch is a bit different. To me,  the RDS in your branch is more up-to-date. Could we merge your RDS to the develop branch in a new branch? This is because I also made a few updates in the RDS. If I pull this branch, it'd be a bit confusing.
>    image
> 
> **Comments**
> 
> 1. It seems like the TCD in this branch isn't valid. Could you check this out?
> 2. I believe User/Proposed should have the `type` key similar to baseline.
> 3. We need the `tilt` key under the `surfaces` b/c the key is required in the `get_opaque_surface_type` function.
> 
> Please let me know if any clarification is needed!

@yunjoonjung-PNNL The RDS which contains the `azimuth_fen_area_binned_dict` is the most current version. We want to create bins of 3 degrees for the surface azimuth dict

**Comment by yunjoonjung-PNNL** [comment]
For some reason, the updated 5-1 TCD is still invalid. Would you be able to double-check it? Or if you update it in the spreadsheet OneDrive, I can debug this.

**Comment by JacksonJ-KC** [comment]
I see the local validation I was running did not have the complete schema as context. The built-in validation is more complete than what I was running. Tests should now pass validation but you will see some issues in the test portion. 

pint.errors.UndefinedUnitError: 'π' is not defined in the unit registry

**Comment by yunjoonjung-PNNL** [comment]
> I see the local validation I was running did not have the complete schema as context. The built-in validation is more complete than what I was running. Tests should now pass validation but you will see some issues in the test portion.
> 
> pint.errors.UndefinedUnitError: 'π' is not defined in the unit registry

I'm curious about this. Did you use π in the spreadsheet? I don't see π in the TCD. In which context did we use π?

**Comment by JacksonJ-KC** [comment]
> > I see the local validation I was running did not have the complete schema as context. The built-in validation is more complete than what I was running. Tests should now pass validation but you will see some issues in the test portion.
> > pint.errors.UndefinedUnitError: 'π' is not defined in the unit registry
> 
> I'm curious about this. Did you use π in the spreadsheet? I don't see π in the TCD. In which context did we use π?

I am not sure but it could potentially be occurring from a conversion between radians and degrees since this rule heavily involves the azimuth - that is my best guess. I believe these are the critical lines of the error message:
File "C:\Users\JacksonJarboe\Documents\Development\RCT\rct229\rule_engine\rule_base.py", line 151, in evaluate
    raw_calc_vals = self.get_calc_vals(context, data)
  File "C:\Users\JacksonJarboe\Documents\Development\RCT\rct229\rulesets\ashrae9012019\section5\section5rule1.py", line 161, in get_calc_vals
    if surface_azimuth_b not in azimuth_fen_area_dict_b:
  File "C:\Users\JacksonJarboe\.virtualenvs\RCT-pOX6qT5Q\lib\site-packages\pint\quantity.py", line 351, in __hash__
    self_base = self.to_base_units()

**Comment by yunjoonjung-PNNL** [comment]
@JacksonJ-KC 

Can you address the following comments?

1. Could you make each baseline (e.g., 90, 180, 270)'s id the same as the ```BASELINE_0```'? For example, change the ```180 Rotation RMD``` in 5-1 (a) to ```RMD 1```? This is because the ids have to be the same because of [this part](https://github.com/pnnl/ruleset-checking-tool/blob/develop/rct229/rule_engine/rule_list_indexed_base.py) (line 155-160). Would @weilixu agree with this point?

2. In 5-1(b), the ```rotation_expected_b``` value has to be False. And, to make it False, the```percent_difference``` should be less than 0.05, but the current value is 1.04. Could you address this?

**Comment by weilixu** [comment]
> @JacksonJ-KC
> 
> Can you address the following comments?
> 
> 1. Could you make each baseline (e.g., 90, 180, 270)'s id the same as the `BASELINE_0`'? For example, change the `180 Rotation RMD` in 5-1 (a) to `RMD 1`? This is because the ids have to be the same because of [this part](https://github.com/pnnl/ruleset-checking-tool/blob/develop/rct229/rule_engine/rule_list_indexed_base.py) (line 155-160). Would @weilixu agree with this point?
> 2. In 5-1(b), the `rotation_expected_b` value has to be False. And, to make it False, the`percent_difference` should be less than 0.05, but the current value is 1.04. Could you address this?

That is correct. The IDs needs to be the same across the RMDs for matching algorithm to work.

**Comment by JacksonJ-KC** [comment]
> @JacksonJ-KC
> 
> Can you address the following comments?
> 
> 1. Could you make each baseline (e.g., 90, 180, 270)'s id the same as the `BASELINE_0`'? For example, change the `180 Rotation RMD` in 5-1 (a) to `RMD 1`? This is because the ids have to be the same because of [this part](https://github.com/pnnl/ruleset-checking-tool/blob/develop/rct229/rule_engine/rule_list_indexed_base.py) (line 155-160). Would @weilixu agree with this point?
> 2. In 5-1(b), the `rotation_expected_b` value has to be False. And, to make it False, the`percent_difference` should be less than 0.05, but the current value is 1.04. Could you address this?

I fixed the RMD ids but the percent difference appears to be an issue with the rule implementation. `min_fen_area_b` should be 24.0 foot ** 2 and `max_fen_area_b` should be 25.0 foot ** 2 but instead `max_fen_area_b` is coming out as 49.0 foot ** 2

**Comment by yunjoonjung-PNNL** [comment]
> > @JacksonJ-KC
> > Can you address the following comments?
> > 
> > 1. Could you make each baseline (e.g., 90, 180, 270)'s id the same as the `BASELINE_0`'? For example, change the `180 Rotation RMD` in 5-1 (a) to `RMD 1`? This is because the ids have to be the same because of [this part](https://github.com/pnnl/ruleset-checking-tool/blob/develop/rct229/rule_engine/rule_list_indexed_base.py) (line 155-160). Would @weilixu agree with this point?
> > 2. In 5-1(b), the `rotation_expected_b` value has to be False. And, to make it False, the`percent_difference` should be less than 0.05, but the current value is 1.04. Could you address this?
> 
> @yunjoonjung-PNNL I fixed the RMD ids but the percent difference appears to be an issue with the rule implementation. `min_fen_area_b` should be 24.0 foot ** 2 and `max_fen_area_b` should be 25.0 foot ** 2 but instead `max_fen_area_b` is coming out as 49.0 foot ** 2

Thanks Jackson. I addressed your comments and all cases passed.

**Comment by weilixu** [comment]
This PR seems outdated. Could you confirm for close? @JacksonJ-KC @yunjoonjung-PNNL

**Comment by yunjoonjung-PNNL** [comment]
> This PR seems outdated. Could you confirm for close? @JacksonJ-KC @yunjoonjung-PNNL

Yes! This PR should be closed as the RMD here is outdated. I believe the 5-1 RMD was merged with the 5-1 ruleset PR in the past.

**Pull request closed without merging**